core/src/main/java/com/sequenceiq/cloudbreak/service/clusterdefinition/ClusterDefinitionService.java

```diff
@@ -237,7 +213,7 @@ public WorkspaceResource resource() {
 
     @Override
     protected void prepareDeletion(ClusterDefinition clusterDefinition) {
```

Comment:
Is there any protection against deleting Default Cluster Definitions?

Reply:
Well, previously, when invoking delete(), a default clusterdefinition's status was set to DEFAULT_DELETED, and some repository methods did not return clusterdefinitions where status was DEFAULT_DELETED. Now, that is changed: for the same repository methods we do not return entities where archived=true. Repo methods where DEFAULT_DELETED was not filtered, I did not change anything. Behaviour should be the same. 
However, thanks for the question, I detected another problem.

Reply:
Now I see you were right, DEFAULTs are indeed deleted - thanks for catching this bug. I add a test to mock to guard against this problem.